explore_data.R

```diff
@@ -23,160 +29,296 @@ source('global.R')
+                                           temp_folder,
+                                           temp_data, sep = '/'),
+                              use.value.labels = T,
+                              to.data.frame = T,
+                              trim.factor.names = T,
+                              trim_values = F,
+                              use.missings = T)
 
-        colnames(temp_dat) <- as.character(temp_dat[1,])
-        temp_dat <- temp_dat[-1,]
 
         # get column names
-        var_names <- as.data.frame(colnames(temp_dat))
-        names(var_names) <- 'var_long'
+        colnames(temp_dat) <- attr(temp_dat,"variable.labels")
+        colnames(temp_dat) <- gsub('.', '', as.character(colnames(temp_dat)), fixed = T)
+        temp_sub <-  temp_dat[, colnames(temp_dat)[colnames(temp_dat) %in% var_names]]
+        colnames(temp_sub) <- tolower(colnames(temp_sub))
+        temp_sub <- as.data.frame(temp_sub, stringsAsFactors = F)
+        data_list[[j]] <- temp_sub
+        } else {
+
+          temp_dat <- as.data.frame(read.sas7bdat(file = paste(path_to_data,
+                                                               temp_folder,
+                                                               temp_data,
+                                                               sep = '/')),
+                                    stringsAsFactors = F)
+          # get column names
+          colnames(temp_dat) <- tolower(colnames(temp_dat))
+          temp_sub <- temp_dat
+          temp_sub <- as.data.frame(temp_sub, stringsAsFactors = F)
+          data_list[[j]] <- temp_sub
+      }
 
-        # missing values per column
-        na_index <- as.data.frame(apply(temp_dat, 2, function(x) length(which(is.na(x)))))
-        names(na_index)[1] <- 'na_counts'
+  }
 
-        # join the two
-        var_data <- cbind(var_names, na_index)
-        var_data$num_rows <- as.numeric(nrow(temp_dat))
-        var_data$num_cols <- as.numeric(ncol(temp_dat))
-        names(var_data) <- c('long_name', 'na_count', 'num_rows', 'num_cols')
-        var_data$per_missing <- round((var_data$na_count/var_data$num_rows)*100, 2)
-        var_data$folder_name <- temp_data
-        var_data$data_name <- temp_data
+  if(length(data_list) > 1) {
+    temp_1 <- data_list[[1]]
+    temp_2 <- data_list[[2]]
+    temp_3 <- data_list[[3]]
 
-        data_list[[j]] <- var_data
 
+    # make colnames the same and join
+    join_key <- Reduce(intersect, list(colnames(temp_1),
+                                       colnames(temp_2),
+                                       colnames(temp_3)))[1]
+
+    # outer join temp1 and temp2
+    temp <- full_join(temp_1, temp_2, by = join_key)
+    temp_full <- full_join(temp, temp_3, by = join_key)
+
+    result_list[[i]] <- temp_full
+
+  } else {
+    result_list[[i]] <- data_list
+
+  }
+
+  print(temp_folder)
+}
+
+length(result_list)
+
+save.image('~/Desktop/survey_temp_readin.RData')
```

Comment:
What's going on with these lines? We shouldn't have hard-coded paths in our code base...

Reply:
ignore explore_data.R those temp objects were just for me to save my place.

Reply:
@benmbrew Okay.